packages/client/hmi-client/src/services/workflow.ts

```diff
@@ -358,3 +358,117 @@ export function updateOutputPort(node: WorkflowNode<any>, updatedOutputPort: Wor
 	if (!outputPort) return;
 	outputPort = Object.assign(outputPort, updatedOutputPort);
 }
+
+// Check if the current-state matches that of the output-state.
+// Note operatorState subsumes the keys of the outputState
+export const isOperatorStateInSync = (
+	operatorState: Record<string, any>,
+	outputState: Record<string, any>
+) => {
+	const hasKey = Object.prototype.hasOwnProperty;
+
+	const keys = Object.keys(outputState);
+	for (let i = 0; i < keys.length; i++) {
+		const key = keys[i];
+		if (!hasKey.call(operatorState, key)) return false;
+		if (!_.isEqual(operatorState[key], outputState[key])) return false;
+	}
+	return true;
```

Comment:
Why not a simpler test on the array?

```suggestion
	return keys.includes(key => 
    hasKey.call(operatorState, key) && _.isEqual(operatorState[key], outputState[key])
  );
```

Reply:
`includes` does not take in a predicate function, and all keys need to meet the conditions, so to do this in one shot I think you will need to do something like "negative of lodash _.some(...)" which I think comes at the expense of comprehension for 1-2 lines fewer code.

Reply:
I see, otherwise `every()` could do that. But fair enough, just curious.